pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -1315,41 +1588,59 @@ void applyDoubleExcitationMinus(
                       const std::size_t i1010, const std::size_t i1011,
                       const std::size_t i1100, const std::size_t i1101,
                       const std::size_t i1110, const std::size_t i1111) {
-            const Kokkos::complex<PrecisionT> v3 = arr(i0011);
-            const Kokkos::complex<PrecisionT> v12 = arr(i1100);
-            arr(i0000) *= e;
-            arr(i0001) *= e;
-            arr(i0010) *= e;
-            arr(i0011) = cr * v3 - sj * v12;
-            arr(i0100) *= e;
-            arr(i0101) *= e;
-            arr(i0110) *= e;
-            arr(i0111) *= e;
-            arr(i1000) *= e;
-            arr(i1001) *= e;
-            arr(i1010) *= e;
-            arr(i1011) *= e;
-            arr(i1100) = sj * v3 + cr * v12;
-            arr(i1101) *= e;
-            arr(i1110) *= e;
-            arr(i1111) *= e;
-        });
+        const Kokkos::complex<PrecisionT> v3 = arr(i0011);
+        const Kokkos::complex<PrecisionT> v12 = arr(i1100);
+        arr(i0000) *= e;
+        arr(i0001) *= e;
+        arr(i0010) *= e;
+        arr(i0011) = cr * v3 - sj * v12;
+        arr(i0100) *= e;
+        arr(i0101) *= e;
+        arr(i0110) *= e;
+        arr(i0111) *= e;
+        arr(i1000) *= e;
+        arr(i1001) *= e;
+        arr(i1010) *= e;
+        arr(i1011) *= e;
+        arr(i1100) = sj * v3 + cr * v12;
+        arr(i1101) *= e;
+        arr(i1110) *= e;
+        arr(i1111) *= e;
```

Comment:
For example. If we pass an array will be possible replace the following code
```suggestion
        for(int i = 0;  i < 16; i++){
           arr(indices[i]) *= e;
        }
        arr[indeces[0B0011] = cr * v3 - sj * v12;
        arr[indeces[0B1100] = sj * v3 + cr * v12;
```

Reply:
The current style clearly describes the unitary matrix.

Reply:
Let's keep the current implement